src/binarySearchTree.js

```diff
@@ -0,0 +1,95 @@
+import BinaryNode from './binarynode'
+
+export default class BinarySearchTree {
+  constructor() {
+    this.length = 0
+    this.root = null
+  }
+
+  insert = (value) => {
+    const node = new BinaryNode(value)
+    if (this.count() == 0) {
```

Comment:
There's a "simpler" recursive solution to insertion, if we add another utility method (in a language with access modifiers, I would make this utility method private, so that it was inaccessible to clients):
```js
insert( value ) {
  if( this.root === null ) {
    this.root = new Node( value )
  } else {
    insertIntoSubtree( value, this.root )
  }
}

insertIntoSubtree( value, treeRoot ) {
  if( value < treeRoot.value ) {
    if( treeRoot.left === null ) {
      treeRoot.left = new Node( value )
    } else {
      insertIntoSubtree( value, treeRoot.left )
    }
  } else {
    // Same thing for right.
  }
}
```
There's probably a more concise way of writing that, but I haven't woken up yet.  😄